### PR TITLE
update api

### PR DESCRIPTION
**Pull request by ex-peter-parker**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [ ] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---

> [!NOTE]
> **Medium Risk**
> Changes request/response shapes and limit validation for a live swap provider, so mismatches with Exolix API expectations could break quoting or cause incorrect limit errors, especially for EVM/token swaps.
> 
> **Overview**
> Updates the Exolix swap integration to use Exolix’s newer network/address-based API parameters: requests now send `networkFrom`/`networkTo` (including an `evmGeneric` path with `evmChainId`) plus optional token `coinAddressFrom`/`coinAddressTo`, and the `/rate` response now requires a `rateId` that is passed through to `/transactions`.
> 
> Replaces the previous currency-code transcription path and the hardcoded $70k USD cap with min/max enforcement directly from Exolix (`minAmount`/`maxAmount` and `withdrawMin`/`withdrawMax`), throwing `SwapBelowLimitError`/`SwapAboveLimitError` accordingly.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f4ddc64a93a4df6158b4a279d0dfe161c9ef94c4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1213596310518725

### CODE REVIEW

File: src/swap/central/exolix.ts [hunk 1]
```diff
@@ -194,47 +239,48 @@ export function makeExolixPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
...
+      networkFromChainId,
+      coinAddressTo,
+      networkTo,
+      networkToChainId,
+      rateType: 'fixed',
+      withdrawalAddress: toAddress,
+      refundAddress: fromAddress,
+      refundExtraId: '',
+      withdrawalExtraId: '',
+      ...amount
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/swap/central/exolix.ts:272]: The concern seems valid here. Perhaps the fix is to use conditional spreads:

```ts
const quoteParams: ExolixQuoteParams = {
  ...(coinAddressFrom != null ? { coinAddressFrom } : {}),
  ...(networkFromChainId != null ? { networkFromChainId } : {})
```

File: src/swap/central/exolix.ts [hunk 2]
```diff
@@ -99,21 +118,35 @@ while true; do
...
 const orderUri = 'https://exolix.com/transaction/'
 const uri = 'https://exolix.com/api/v2/'
 
 const expirationMs = 1000 * 60
 
 const asRateResponse = asObject({
   minAmount: asNumber,
+  maxAmount: asNumber,
   withdrawMin: asOptional(asNumber, 0),
+  withdrawMax: asOptional(asNumber, 0),
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/swap/central/exolix.ts]: Solution for this is to keep `withdrawMax` optional with no default and case the check under the condition where `withdrawMax` is present.

File: src/swap/central/exolix.ts [hunk 3]
```diff
@@ -367,71 +422,6 @@ export function makeExolixPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
...
-        rates = asRatesResponse(reply)
-      } catch (e) {
-        log.error('Error fetching rates', String(e))
-        throw new Error('Error fetching rates')
-      }
-
-      const { exchangeRate } = rates.data[0]
-      if (exchangeRate == null) throw new SwapCurrencyError(swapInfo, request)
-
-      const usdValue = mul(exchangeAmount, exchangeRate)
```
Comment (samholmes) [inline src/swap/central/exolix.ts]: **Warning:** This PR materially changes Exolix quote behavior (new request shape, max/min handling, and 422 parsing) without corresponding automated tests.\n\n**Recommendation:** Add targeted tests for `quoteFor: from/to`, missing `withdrawMax`, and 422 minimum responses to prevent regressions.

File: src/swap/central/exolix.ts [hunk 4]
```diff
@@ -99,21 +118,35 @@ while true; do
...
+}
+
 const orderUri = 'https://exolix.com/transaction/'
 const uri = 'https://exolix.com/api/v2/'
 
 const expirationMs = 1000 * 60
 
 const asRateResponse = asObject({
   minAmount: asNumber,
+  maxAmount: asNumber,
```
Comment (samholmes) [inline src/swap/central/exolix.ts:143]: **Issue:** The 422 fallback uses `asMaybe(asRateResponse)`, but this cleaner now requires fields like `maxAmount` and `rateId`. If Exolix returns a partial 422/minimum payload, parsing fails and this path throws `SwapCurrencyError` instead of returning a precise limit error.

**Recommendation:** Use a dedicated looser cleaner for 422 limit payloads, and reserve `asRateResponse` for successful quote payloads.

File: src/swap/central/exolix.ts [hunk 5]
```diff
@@ -254,26 +310,25 @@ export function makeExolixPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
...
-      networkTo: quoteParams.coinToNetwork,
-      amount: quoteAmount,
-      withdrawalAddress: toAddress,
-      withdrawalExtraId: '',
-      refundAddress: fromAddress,
-      refundExtraId: '',
-      rateType: 'fixed'
+    const exchangeParams: ExolixQuoteParams = {
+      ...quoteParams,
+      rateId: rateResponse.rateId
```
Comment (samholmes) [inline src/swap/central/exolix.ts:332]: **Warning:** This change removes the prior local `$70k` USD guardrail. If that cap was a product/risk requirement, behavior has changed and large swaps now rely solely on partner-provided limits.

**Recommendation:** Confirm policy intent and keep a local cap if still required.

File: src/swap/central/exolix.ts [hunk 6]
```diff
@@ -367,71 +422,6 @@ export function makeExolixPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
       const fixedOrder = await getFixedQuote(newRequest, userSettings)
       const fixedResult = await makeSwapPluginQuote(fixedOrder)
```
Comment (samholmes) [inline src/swap/central/exolix.ts:427]: **Warning:** This PR materially changes Exolix quote behavior (new request shape, max/min handling, and 422 parsing) without corresponding automated tests.

**Recommendation:** Add targeted tests for `quoteFor: from/to`, missing `withdrawMax`, and 422 minimum responses to prevent regressions.

Comment (samholmes) [inline src/swap/central/exolix.ts:427]: From the API requirements angle, this PR does move the integration in the right direction in a few meaningful ways: it updates the request shape to use chain-aware identifiers (`evmGeneric` plus chain IDs for EVM assets), includes token contract addresses, and adds explicit destination-side quoting support via `quoteFor: 'to'`. Those are all improvements toward the chain/token identification and bi-directional quoting requirements.

The highest-priority remaining concern is whether the new destination-side quoting path is fully reliable in practice. Because `quoteFor: 'to'` is new here and limit handling now depends on `withdrawMin` / `withdrawMax`, I would want confidence that Exolix always returns the necessary limit fields or that the plugin tolerates missing fields without manufacturing false over-limit failures.

Beyond that, the other material API-requirements items are provider capabilities that matter to user experience and ops, but are not really proven by this plugin diff: an unauthenticated status page, a transaction-status API by `orderId`, and reporting data that can be matched back to Edge orders. I would not block this PR on every last detail of error-shape perfection or extra hardening if the Exolix team has already confirmed those behaviors, but before treating the integration as compliant with the API requirements, I would want those core capabilities explicitly confirmed somewhere in the PR or supporting notes.

**Review by samholmes — CHANGES_REQUESTED** [review]

File: src/swap/central/exolix.ts (hunk 5 above)
Comment (ex-peter-parker) [reply src/swap/central/exolix.ts:332]: I confirm this — it was necessary for the old functionality; now all limits are calculated automatically.

File: src/swap/central/exolix.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ex-peter-parker) [reply src/swap/central/exolix.ts]: fixed

File: src/swap/central/exolix.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ex-peter-parker) [reply src/swap/central/exolix.ts:272]: fixed
